glide-core/src/client/value_conversion.rs

```diff
@@ -567,6 +657,7 @@ pub(crate) fn expected_type_for_cmd(cmd: &Cmd) -> Option<ExpectedReturnType> {
             }
         }
         b"LOLWUT" => Some(ExpectedReturnType::Lolwut),
+        b"XREAD" => Some(ExpectedReturnType::XreadReturnType),
```

Comment:
Can you try this trick?
```suggestion
        b"XREAD" => Some(ExpectedReturnType::Map(&ExpectedReturnType::Map(&ExpectedReturnType::Array))),
```

It is similar to https://github.com/aws/glide-for-redis/blob/fa3c80abfae3df63a4767a9afaec36deeb85587e/glide-core/src/client/value_conversion.rs#L681-L682